zulipterminal/ui_tools/views.py

```diff
@@ -411,17 +411,30 @@ def __init__(self, width: int, view: Any) -> None:
                                               header=search_box)
 
     @asynch
-    def update_user_list(self, search_box: Any, new_text: str) -> None:
-        if not self.view.controller.editor_mode:
+    def update_user_list(self, search_box: Any=None,
+                         new_text: str="",
+                         user_list: Any=None) -> None:
+
+        assert ((user_list is None and search_box is not None and
+                 new_text != "") or
+                (user_list is not None and search_box is None and
+                 new_text == "") or
+                (user_list is None and search_box is not None and
+                 new_text == ""))
+
+        if not self.view.controller.editor_mode and not user_list:
             return
         # wait for any previously started search to finish to avoid
         # displaying wrong user list.
         self.search_lock.acquire()
+        if user_list:
+            self.view.users = user_list
         users = self.view.users.copy()
         users_display = users.copy()
```

Comment:
I think this is ok to stay for a v1 of this new feature, but I think it'd be good to see these copies avoided for the `user_list` case, and simplified further using a comprehension otherwise.

Reply:
Agreed. Will keep this in mind :+1: